app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -69,6 +70,26 @@ def datastores(opts = {})
       end
     end
 
+    def storage_pods(opts = {})
+      if opts[:storage_pod]
+        begin
+          dc.storage_pods.get(opts[:storage_pod])
+        rescue RbVmomi::VIM::InvalidArgument
+          {}
+        end
+      else
+        begin
+          name_sort(dc.storage_pods.all())
+        rescue RbVmomi::VIM::InvalidArgument
```

Comment:
Kinda confused at these rescues, why does one of them return a hash & the other returns an array? I'd assume the first one would return an array containing the storage pod when you pass the option, and the second one would return an array of all storage pods.